ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,137 @@
+import json
+import logging
+from json import JSONDecodeError
+import requests
+
+from ocp_resources.route import Route
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.utils import TimeoutExpiredError, TimeoutSampler
+from ocp_utilities.infra import get_client
+
+TIMEOUT_10MIN = 10 * 60
+
+LOGGER = logging.getLogger(__name__)
+
+class Prometheus(object):
+    """
+    For accessing Prometheus cluster metrics
+
+    Prometheus HTTP API doc:
+    https://prometheus.io/docs/prometheus/latest/querying/api/
+
+    Argument for query method should be the entire string following the server address
+        e.g.
+        prometheus = Prometheus()
+        up = prometheus.query("/api/v1/query?query=up")
+    """
+
+    def __init__(
+        self,
+        namespace="openshift-monitoring",
+        resource_name="prometheus-k8s",
+        client=None,
+    ):
+        self.namespace = namespace
+        self.resource_name = resource_name
+        self.client = get_client()
```

Comment:
We should use the user client if we got one, else generate the client.

Reply:
ack